cli/chaosli/cmd/create.go

```diff
@@ -0,0 +1,485 @@
+		hrPair := v1beta1.HostRecordPair{}
+
+		hrPair.Hostname = getInput("Specify a hostname to target",
+			"When your target makes a DNS request for this hostname; the disruption will make sure the value you specify is returned, rather than the real record.",
+			true,
+		)
+		hrPair.Record.Type, _ = selectInput("the type of DNS record to inject",
+			[]string{"A", "CNAME"},
+			"We only support these two types of DNS requests for now. An A record request gets back an IP for a hostname, while a CNAME request maps an alias domain name to the canonical name.")
+		helpText := "We're specifying an A record, so the value should be an IP address. You can specify multiple IP addresses, if desired. Simply delimit them with commas, no whitespace! The disruption will round-robin between the options."
+
+		if hrPair.Record.Type == "CNAME" {
+			helpText = "We're specifying a CNAME record, so the value should be a hostname to redirect to."
+		}
+
+		hrPair.Record.Value = getInput("What value would you like to inject into this DNS record?", helpText, true)
+
+		return hrPair
+	}
+
+	fmt.Println("Let's specify a DNS record to inject!")
+
+	spec := v1beta1.DNSDisruptionSpec{}
+	spec = append(spec, getHostRecordPair())
+
+	for confirmOption("Would you like to override another DNS record?", "") {
+		spec = append(spec, getHostRecordPair())
+	}
+
+	return spec
+}
+
+func getDiskPressure() *v1beta1.DiskPressureSpec {
+	if !confirmKind("Disk Pressure", "Applies IO pressure to the target") {
+		return nil
+	}
+
+	spec := &v1beta1.DiskPressureSpec{}
+
+	spec.Path = getInput(
+		"Specify a path to apply IO pressure to",
+		"Specify a specific mount point to target a specific disk",
+		true,
+	)
+
+	if confirmOption("Would you like to apply read pressure?", "This applies read-based IO pressure (check the docs)") {
+		readBPS, _ := strconv.Atoi(getInput("Specify the target amount of pressure, in bytes per second.", "check the docs", false))
+		spec.Throttling.ReadBytesPerSec = &readBPS
+	}
+
+	if confirmOption("Would you like to apply write pressure?", "This applies write-based IO pressure (check the docs)") {
+		writeBPS, _ := strconv.Atoi(getInput("Specify the target amount of pressure, in bytes per second.", "check the docs", false))
+		spec.Throttling.WriteBytesPerSec = &writeBPS
+	}
+
+	return spec
+}
+
+func getCPUPressure() *v1beta1.CPUPressureSpec {
+	if confirmKind("CPU Pressure", "Applies CPU pressure to the target") {
+		return &v1beta1.CPUPressureSpec{}
+	}
+
+	return nil
+}
+
+func getNodeFailure() *v1beta1.NodeFailureSpec {
+	if !confirmKind("Node Failure", "This will either shutdown or restart the targeted node (or node hosting the targeted pod)") {
+		return nil
+	}
+
+	spec := &v1beta1.NodeFailureSpec{}
+	spec.Shutdown = confirmOption("Would you like to shutdown the node permanently?",
+		"Choosing yes will terminate the VM completely, and a new node will be scheduled. If you don't enable this, we will just restart the target node.")
```

Comment:
I dont know if new nodes are scheduled by default. The team would require a nodegroup, not sure if we have any teams at Datadog that manage their own nodes.

Reply:
That's true, I am describing DD specific behavior, good point